js/coinflex.js

```diff
@@ -0,0 +1,1885 @@
+        //                 "marketCode": "SHIB-USD-SWAP-LIN",
+        //                 "matchQuantity": "313546",
+        //                 "matchPrice": "0.00002161",
+        //                 "total": "6.77572906",
+        //                 "orderMatchType": "TAKER",
+        //                 "fees": "0.00542058",
+        //                 "feeInstrumentId": "USD",
+        //                 "orderId": "1002109741555",
+        //                 "side": "BUY",
+        //                 "clientOrderId": "1651342025382"
+        //             }
+        //         ]
+        //     }
+        //
+        const trades = this.safeValue (response, 'data', {});
+        return this.parseTrades (trades, market, since, limit, params);
+    }
+
+    parseTrade (trade, market = undefined) {
+        //
+        // fetchTrades
+        //
+        //     {
+        //         "matchId": "304734619669458401",
+        //         "matchQuantity": "0.012",
+        //         "matchPrice": "38673",
+        //         "side": "BUY",
+        //         "matchTimestamp": "1651281046230"
+        //     }
+        //
+        // fetchMyTrades
+        //
+        //     {
+        //         "matchId": "8375163007067643787",
+        //         "matchQuantity": "313546",
+        //         "matchPrice": "0.00002161",
+        //         "side": "BUY",
+        //         "matchTimestamp": "1651342025862",
+        //         "marketCode": "SHIB-USD-SWAP-LIN",
+        //         "total": "6.77572906",
+        //         "orderMatchType": "TAKER",
+        //         "fees": "0.00542058",
+        //         "feeInstrumentId": "USD",
+        //         "orderId": "1002109741555",
+        //         "clientOrderId": "1651342025382"
+        //     }
+        //
+        // trades from order-object
+        //
+        //     {
+        //         "8375163007067827477": {
+        //             "matchQuantity": "334016",
+        //             "matchPrice": "0.00002089",
+        //             "timestamp": "1651410712318",
+        //             "orderMatchType": "TAKER"
+        //         }
+        //     }
+        //
+        const marketId = this.safeString (trade, 'marketCode');
+        market = this.safeMarket (marketId, market);
+        let id = undefined;
+        let timestamp = undefined;
+        let priceString = undefined;
+        let amountString = undefined;
+        let side = undefined;
+        let cost = undefined;
+        let fee = undefined;
+        let takerOrMakerRaw = undefined;
+        const keys = Object.keys (trade);
+        const length = keys.length;
+        if (length === 1) {
+            id = keys[0];
+            const tradeData = trade[id];
+            amountString = this.safeString (tradeData, 'matchQuantity');
+            priceString = this.safeString (tradeData, 'matchPrice');
+            cost = Precise.stringMul (amountString, priceString);
```

Comment:
`safeTrade` does that calculation for us

Reply:
👍